### PR TITLE
Move bench files

### PR DESCRIPTION
**Pull request by pedrohba1**
Bench files were being pulled when using as a dependency so I moved them to tests folder where this won't happen. Making the life of anyone using flat_files_decoder as dependency easier on the internet download. Also, I've added more details about this code on README.md

@severiano-sisneros can you check the TODO's in this code? I've written them in the key parts of the doc that I was not entirely sure on.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,25 +1,87 @@
 # Flat files decoder for firehose
 
-## Usage
+[![CI status](https://github.com/semiotic-ai/flat-files-decoder/workflows/ci/badge.svg)][gh-ci]
+
+<!-- TODO: Seve please checkout if what I wrote makes sense -->
+this crate is designed to decompress and decode headers from [binary files, which are called flat files,](https://github.com/streamingfast/firehose-ethereum/blob/develop/proto/sf/ethereum/type/v2/type.proto) generated from Firehose. Flat files store all information necessary to reconstruct the transaction and receipt tries. It also checks the validity of 
+receipt roots and transaction roots present in the block headers by recalculating them via the block body data. Details of the implementation can be found [here](https://github.com/streamingfast/dbin?tab=readme-ov-file)
```
Comment (severiano-sisneros) [inline Readme.md:7]: > It also checks the validity of receipt roots and transaction roots present in the block headers by recalculating them via the block body data.

I might add one line after this: "This check ensures that receipt logs and transaction data stored in the flat files are internally consistent with the block headers also stored in the flat files."

File: Readme.md [hunk 2]
```diff
@@ -1,25 +1,87 @@
...
+
+This tool was first presented as a mean to enhance the performance and verifiability of The Graph protocol. However,
+it turns out it could be used as a solution for EIP-4444 problem of full nodes stopping to provide historical data over one year.
+The idea is that the flat files that this crate can decode could also be used as an archival format similar to era1 files, specially
+if they can be verified.
```
Comment (severiano-sisneros) [inline Readme.md:12]: > if they can be verified. 

Maybe reword and add a little to this sentence:
" ... if the transaction and receipt data stored in the flat file can be verified to be part of Ethereum's canonical history."

Comment (severiano-sisneros) [inline Readme.md:12]: It might be worth adding a line about how we verify:

"See our ethresear.ch post, [here](https://ethresear.ch/t/using-the-graph-to-preserve-historical-data-and-enable-eip-4444/), and our pre-merge solution, implemented [here](https://github.com/semiotic-ai/header_accumulator) to see how we verify that the flat files are consistent with Ethereum's canonical history.

File: Readme.md [hunk 3]
```diff
@@ -1,25 +1,87 @@
...
+This will store the block headers as json format in the output folder. 
+By passing `--headers-dir` a folder of assumed valid block headers can be provided to compare
+with the input flat files. Valid headers can be pulled from the [sync committee subprotocol](https://github.com/ethereum/annotated-spec/blob/master/altair/sync-protocol.md) for post-merge data.
+
+<!-- TODO: once the header_accumulator is made public, link it here -->
+**NOTICE:**For pre-merge data another approach using [header accumulators](https://github.com/ethereum/portal-network-specs/blob/8ad5bc33cb0d4485d2eab73bf2decc43e7566a8f/history-network.md#the-header-accumulator) is necessary since
+sync committees will not provide these headers.
+
+## Goals
+<!-- TODO: Any other goals I should add? -->
```
Comment (severiano-sisneros) [inline Readme.md:74]: Maybe a note about how we hope Indexers/Firehose operators will use this, e.g.:

"Our goal is to provide The Graph's Indexers the tools to trustlessly share flat files with cryptographic guarantees that the data in the flat files is part of the canonical history of the Ethereum blockchain, enabling Indexers to quickly sync all historical data and begin serving data with minimal effort."

**Comment by pedrohba1** [comment]
@severiano-sisneros sending this message to see if github extension for slack works to notify you.
Also, checkout the doc I added to receipts, if it makes sense. I wanted to clarify a bit of the inner workings of the tool using reth

**Comment by severiano-sisneros** [comment]
> @severiano-sisneros sending this message to see if github extension for slack works to notify you. Also, checkout the doc I added to receipts, if it makes sense. I wanted to clarify a bit of the inner workings of the tool using reth

@pedrohba1 doesn't look like I was notified in slack...

File: src/receipts/mod.rs [hunk 4]
```diff
@@ -40,8 +56,26 @@ fn calc_receipt_root(block: &Block) -> Result<B256, ReceiptError> {
     Ok(ordered_trie_root_with_encoder(&receipts, encoder))
 }
 
+/// Encodes full rceipts using [RLP serialization](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp)
+///
+/// For blocks before the Byzantium fork, it uses a specific RLP encoding that includes the receipt's header length values, state root,
```
[the remark this answers is not in the document]
Comment (pedrohba1) [reply src/receipts/mod.rs:60]: > We're actually being a little "hacky" here. The `encode_inner` function is imported from reth. Reth doesn't support receipts before the Byzantium fork, see [here](https://github.com/paradigmxyz/reth/issues/853) so we make the changes here to support pre-Byzantium receipts.

After seeing what is done in Geth I think I will give an attempt into implementing in Reth. Let's approve this for now so I can work on the next step
